packages/components/src/theme/ThemeProvider.tsx

```diff
@@ -47,8 +48,13 @@ export function ThemeProvider({
 
   const [value, setValue] = useState<ThemeContextValue | null>(null);
 
+  const themeKeyOverride = useMemo(() => getThemeKeyOverride(), []);
+
   const [selectedThemeKey, setSelectedThemeKey] = useState<string>(
-    () => getThemePreloadData()?.themeKey ?? DEFAULT_DARK_THEME_KEY
+    () =>
+      themeKeyOverride ??
+      getThemePreloadData()?.themeKey ??
+      DEFAULT_DARK_THEME_KEY
```

Comment:
May as well wrap this all in a function, `getDefaultTheme` or `getDefaultSelectedTheme`. Then this just becomes.
```
... useState<string>(getDefaultSelectedTheme)
```